### PR TITLE
Raise APIError from close_channel

### PR DESCRIPTION
**Pull request by yuntai**

### CODE REVIEW

File: src/ln/channel.rs [hunk 1]
```diff
@@ -2755,18 +2755,18 @@ impl Channel {
...
 			}
 		}
 		if self.channel_state & BOTH_SIDES_SHUTDOWN_MASK != 0 {
-			return Err(HandleError{err: "Shutdown already in progress", action: None});
+			return Err(APIError::APIMisuseError{err: "Shutdown already in progress"});
 		}
 		assert_eq!(self.channel_state & ChannelState::ShutdownComplete as u32, 0);
 		if self.channel_state & (ChannelState::PeerDisconnected as u32) == ChannelState::PeerDisconnected as u32 {
-			return Err(HandleError{err: "Cannot begin shutdown while peer is disconnected, maybe force-close instead?", action: None});
+			return Err(APIError::APIMisuseError{err: "Cannot begin shutdown while peer is disconnected, maybe force-close instead?"});
```
Comment (TheBlueMatt) [inline src/ln/channel.rs]: Hmm, can we really call this "Misuse"? Maybe some less-user-blaming enum entry in APIError?

File: src/ln/channel.rs [hunk 2]
```diff
@@ -2755,18 +2755,18 @@ impl Channel {
 
 	/// Begins the shutdown process, getting a message for the remote peer and returning all
 	/// holding cell HTLCs for payment failure.
-	pub fn get_shutdown(&mut self) -> Result<(msgs::Shutdown, Vec<(HTLCSource, [u8; 32])>), HandleError> {
+	pub fn get_shutdown(&mut self) -> Result<(msgs::Shutdown, Vec<(HTLCSource, [u8; 32])>), APIError> {
 		for htlc in self.pending_outbound_htlcs.iter() {
 			if htlc.state == OutboundHTLCState::LocalAnnounced {
-				return Err(HandleError{err: "Cannot begin shutdown with pending HTLCs, call send_commitment first", action: None});
+				return Err(APIError::APIMisuseError{err: "Cannot begin shutdown with pending HTLCs. Process pending events first"});
```
Comment (TheBlueMatt) [inline src/ln/channel.rs]: I *think* we can debug_assert!(false) here.

Comment (yuntai) [reply src/ln/channel.rs]: but the client can call close_channel() anytime?

Comment (TheBlueMatt) [reply src/ln/channel.rs]: Oh, oops, confused myself as to what LocalAnnounced meant. Forgot that it implied "included in sent commitment_signed". Indeed, I was mistaken.

File: src/ln/channel.rs (hunk 1 above)
Comment (yuntai) [reply src/ln/channel.rs]: added ChannelUnavailalbe enum entry for APIError.

File: src/ln/channelmanager.rs [hunk 3]
```diff
@@ -439,7 +439,7 @@ impl ChannelManager {
 						(res, chan_entry.get().get_their_node_id(), Some(chan_entry.remove_entry().1))
 					} else { (res, chan_entry.get().get_their_node_id(), None) }
 				},
-				hash_map::Entry::Vacant(_) => return Err(HandleError{err: "No such channel", action: None})
+				hash_map::Entry::Vacant(_) => return Err(APIError::APIMisuseError{err: "No such channel"})
```
Comment (TheBlueMatt) [inline src/ln/channelmanager.rs]: Shouldnt this also be ChannelUnavailable (after all, there are always going to be races on the client's side where a channel gets (force-)closed between when they decide to close it and when they call close_channel).

**Pull request merged**